tfe/resource_tfe_registry_module.go

```diff
@@ -171,3 +185,15 @@ func resourceTFERegistryModuleImporter(d *schema.ResourceData, meta interface{})
 
 	return []*schema.ResourceData{d}, nil
 }
+
+func checkModuleExists(ctx context.Context, tfeClient *tfe.Client, organization, name, module_provider string, doneCh chan bool) {
+	for {
+		_, err := tfeClient.RegistryModules.Read(ctx, organization, name, module_provider)
+		if err != nil {
+			time.Sleep(time.Duration(100) * time.Millisecond)
+			continue
+		}
+		doneCh <- true
+		break
+	}
+}
```

Comment:
since we're waiting for up to 5 minutes for the module to finish importing, it might be a good to make the sleep time a little longer here so we aren't sending so many requests to TFC/TFE. maybe something like 5-10 seconds? or possibly a very simple exponential backoff of 1 second, 2, 4, etc. what do you think?

Reply:
Good call! As a matter of fact I found out that the plugin SDK provides this functionality on its own (https://godoc.org/github.com/hashicorp/terraform-plugin-sdk/helper/resource#Retry). 

I changed the code to use that instead.